src/query/v2/plan/operator.cpp

```diff
@@ -257,13 +257,30 @@ class DistributedCreateNodeCursor : public Cursor {
 bool Once::OnceCursor::Pull(Frame &, ExecutionContext &context) {
   SCOPED_PROFILE_OP("Once");
 
-  if (!did_pull_) {
-    did_pull_ = true;
+  if (pull_count_ < 1) {
+    pull_count_++;
     return true;
   }
   return false;
 }
 
+void Once::OnceCursor::PullMultiple(MultiFrame &multi_frame, ExecutionContext &context) {
+  SCOPED_PROFILE_OP("OnceMF");
+
+  auto iterator_for_valid_frame_only = multi_frame.GetValidFramesConsumer();
+  auto first_it = iterator_for_valid_frame_only.begin();
+  MG_ASSERT(first_it != iterator_for_valid_frame_only.end());
+  if (pull_count_ < 1) {
+    auto *memory_resource = multi_frame.GetMemoryResource();
+    auto &frame = *first_it;
+    frame.MakeValid();
+    for (auto &value : frame.elems()) {
+      value = TypedValue{memory_resource};
+    }
```

Comment:
I assume you get this from my experiment branch. I did a mistake there. The `OnceCursor` shouldn't modify the `Frame` in the "old approach" because of the optional match. I think we will have other issues with the optional match, but this is for sure will be an issue. So maybe just call `MakeValid` on the first `Frame` of the `MultiFrame` (without checkout if it is valid or not, we shouldn't care in my opinion)

Reply:
Here you still modify the values on the `Frame` which doesn't work well with the optional cursor. I think lines 277-279 should be deleted:
```suggestion
    auto &frame = *first_it;
    frame.MakeValid();
```